orangecontrib/bio/widgets3/OWFeatureSelection.py

```diff
@@ -49,7 +49,12 @@ def score_fold_change(a, b, axis=0):
     """
     mean_a = np.nanmean(a, axis=axis)
     mean_b = np.nanmean(b, axis=axis)
-    return mean_a / mean_b
+    res = mean_a / mean_b
+    warning = None
+    if np.any(res < 0):
+        res[res < 0] = float("nan")
+        warning = "Negative fold change scores were ignored. You should use another scoring method."
+    return res, warning
 
```

Comment:
I think it should just raise an ValueError if either a or b contain any negative values.